packages/markdown/remark/src/types.ts

```diff
@@ -68,6 +68,12 @@ export interface MarkdownMetadata {
 	html: string;
 }
 
+export interface MarkdownVFile extends VFile {
+	data: {
+		__astroHeadings?: MarkdownHeading[];
+	};
```

Comment:
```suggestion
	data: {
		__astroHeadings?: MarkdownHeading[];
		astro: {
      frontmatter: Record<string, any>;
    }
	};
```
Nit: if we're typing our VFile data, we can add a type for Astro frontmatter as well! See [Astro core type](https://github.com/withastro/astro/blob/main/packages/astro/src/%40types/astro.ts#L1506)

Reply:
Hmm. This type is intentionally only used internally as it’s kind of a private API. Are you thinking someone might import `MarkdownVFile` and use it in their custom plugins as a convenience when injecting frontmatter?

Reply:
Hm, honestly wasn't sure. We can definitely ignore since `astro.frontmatter` is superfluous for the type's use case.